tests_integration/test_snowpark_function.py

```diff
@@ -0,0 +1,417 @@
+from __future__ import annotations
+
+import re
+from typing import Any, List, Dict
+
+import pytest
+from snowflake.connector import SnowflakeConnection
+from syrupy import SnapshotAssertion
+
+from tests_integration.testing_utils.file_utils import replace_text_in_file
+from tests_integration.conftest import runner
+from tests_integration.snowflake_connector import create_database, snowflake_session
+from tests_integration.testing_utils.sql_utils import sql_test_helper
+from tests_integration.testing_utils.naming_utils import object_name_provider
+from tests_integration.testing_utils.working_directory_utils import (
+    temporary_working_directory,
+)
+
+from tests_integration.conftest import SnowCLIRunner
+from tests_integration.testing_utils.assertions.test_file_assertions import (
+    assert_that_current_working_directory_contains_only_following_files,
+    assert_that_file_content_is_equal_to_snapshot,
+)
+from tests_integration.testing_utils.assertions.test_result_assertions import (
+    assert_that_result_is_successful_and_has_no_output,
+    assert_that_result_is_successful_and_output_contains,
+    assert_that_result_is_successful,
+    assert_that_result_is_successful_and_json_output_contains_value_at_path,
+)
+from tests_integration.testing_utils.naming_utils import ObjectNameProvider
+from tests_integration.testing_utils.sql_utils import SqlTestHelper
+
+
+@pytest.mark.integration
+def test_snowpark_function_flow(_test_steps):
+    _test_steps.assert_that_no_functions_are_in_snowflake()
+    _test_steps.assert_that_no_files_are_staged_in_snowflake()
+
+    _test_steps.snowpark_function_list_should_return_no_data()
```

Comment:
We should test it if functions are in scope of database, if yes this check is not needed

Reply:
Can you explain what do you mean by `if functions are in scope of database`?

Current assertions in the test case check exactly the same what I check manually when testing the tool:
* empty start at the beginning (both presented by Snowflake SQL output and presented by our tool)
    * I can agree that maybe I don't have to execute SQLs here, but still IMO it's worth to check correct output of `function list`.
* successful execution of commands
* correct state of local files after commands which can change it
* correct state in Snowflake after commands which can change it (checked using SQL)
* correct presentation and usage of functions by our tool (list, describe, execute) after commands which can change functions state